prometheusbp/promtest/testutil.go

```diff
@@ -10,24 +10,36 @@ import (
 
 // PrometheusMetricTest stores information about a metric to use for testing.
 type PrometheusMetricTest struct {
-	tb        testing.TB
-	metric    prometheus.Collector
-	name      string
-	initValue float64
-	labels    prometheus.Labels
+	tb             testing.TB
+	metric         prometheus.Collector
+	name           string
+	initValue      float64
+	initHistoCount int
+	labels         prometheus.Labels
 }
 
 // CheckDelta checks that the metric value changes exactly delta from when Helper was called.
 func (p *PrometheusMetricTest) CheckDelta(delta float64) {
 	p.tb.Helper()
 
-	got := p.getValue()
+	got, _ := p.getValues()
 	got -= float64(p.initValue)
 	if got != delta {
 		p.tb.Errorf("%s metric delta: wanted %v, got %v", p.name, delta, got)
 	}
 }
 
+// CheckHistoCountDelta checks that the number of samples of histogram is the exactly delta from when Helper was called.
+func (p *PrometheusMetricTest) CheckHistoCountDelta(delta int) {
+	p.tb.Helper()
+
+	_, got := p.getValues()
```

Comment:
I don't really like conflating getValues to return a second argument all the time...

it makes it look like errors are being disccarded.

Maybe make a new getSampleCount helper?

If not, `getValues -> valueAndSamples` maybe to make it clear at the call site?